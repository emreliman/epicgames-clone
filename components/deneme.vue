<template>
    <h1>Emre</h1>
</template>

<script>
export default {
    
}
</script>