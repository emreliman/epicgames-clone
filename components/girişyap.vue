<template>
    <div class="giriş-card">
        <div class="giriş-logo">
          <img src="@/assets/epicgames-logo.png" alt="logo" />
        </div>
        <p
          style="
            color: rgba(255, 255, 255, 0.75);
            margin-left: 30px;
            font: Brutal, sans-serif;
            font-size: 13px;
          "
        >
          BİR EPIC GAMES HESABI ILE GIRIŞ YAP
        </p>

        <div class="giriş-görünenad">
          <input v-model="email" placeholder="*E-posta Adresi" required ><p style="margin-top:25px ;transform: translateX(-55px);">Gerekli</p> </input>
        </div>

        <div class="giriş-şifre">
          <input v-model="password" placeholder="*Şifre " type="password" required ><p style="margin-top:25px ;transform: translateX(-55px);" >Gerekli</p></input>
         
        </div>
         <p id="errormessage" v-if="error">{{message}} </p>
        <div class="giriş-teklifler">
          <input type="checkbox" />
          <p style="margin-top: 18px; margin-right: 110px">Beni hatırla</p>
          <p style="margin-top: 18px">Şifreni Mİ Unuttun?</p>
        </div>

        <button @click="signIn" class="giriş-devamet">HEMEN GİRİŞ YAP</button>
        <p
          style="
            text-align: center;
            color: rgba(255, 255, 255, 0.815);
            font-size: 12px;
          "
        >
          Gizlilik Politikası
        </p>
        <p
          style="
            color: rgba(244, 244, 244, 0.5);
            text-align: center;
            font-size: 12px;
          "
        >
          Epic Games hesabın yok mu? Kayıt Ol
        </p>
        <p
          style="
            color: rgba(244, 244, 244, 0.5);
            text-align: center;
            font-size: 12px;
            margin-top: -5px;
          "
        >
          Tüm kaydolma seçeneklerine geri dön
        </p>
        
      </div>

</template>
<script>
import {mapState,mapActions} from 'vuex';
import config from "../assets/config.js"
import firebase from 'firebase/app';
require('firebase/auth')
export default {
  data(){
    return{
      email:'',
      password:'',
      usern:null,
      error:false,
      message:null,
    }
  },
  created(){
   
   
  //  firebase.auth().onAuthStateChanged(user=>{
  //       this.usern=user
  //       this.aUpdateUser(this.usern);
  //     })
   


  },
 
  

  methods:{
    signIn(e){
      if (!firebase.apps.length) {
            firebase.initializeApp(config);
                                       }
            else {
              firebase.app(); 
                 }

      e.preventDefault();
      firebase.auth().signInWithEmailAndPassword(this.email,this.password).then(()=>{
        console.log('all Good!');
        this.aUpdateUser(this.email);
        this.$root.$emit('logg',this.email);
        this.error=false;
      }
      ).catch(err=>{
        this.error=true;
        this.message=err.message;
      })
      
      
      
    },
    ...mapActions([
      'aUpdateUser'
    ]),
    
  }
}
</script>
<style  scoped>
#errormessage{
  color: red;
  font-size: 12px;
  margin-left: 30px;

}
</style>