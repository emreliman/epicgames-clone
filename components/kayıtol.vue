<template>
  <div class="kayıtol-card">
    <div class="kaytol-logo">
      <img src="@/assets/epicgames-logo.png" alt="logo" />
    </div>
    <div class="kayıtol-droplist">
      <p
        style="
              color: #f4f4f4;
              font-size: 12px;
              font-family: Brutal, sans-serif;
              margin-left: 32px;
            "
      >
        KAYIT OL
      </p>

      <button class="kayıtol-dropbtn">
        Turkey
        <svg
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
          aria-hidden="true"
          style="width: 2rem; height: 2rem; float: right"
        >
          <path d="M7 10l5 5 5-5z"></path>
        </svg>
      </button>
      <div class="kayıtol-drop-content">
        <!--
            <ul>
              <li>Turkey</li>
              <li>United States</li>
              <li>United Kingdom</li>
            </ul>
            -->
      </div>
    </div>
    <div class="kayıtol-adsoyad">
      <input placeholder="*Ad" required />
      <input placeholder="*Soyadı" required />
    </div>
    <div class="kayıtol-görünenad">
      <input placeholder="*Görünen Ad" required />

      <input v-model="email" placeholder="*E-posta Adresi" required />
    </div>

    <div class="kayıtol-şifre">
      <input
        v-model="password"
        placeholder="*Şifre "
        type="password"
        required
      />
    </div>
    <div class="kayıtol-teklifler">
      <input type="checkbox" />
      <p>
        Epic Games'ten haberler, anketler ve özel teklifler almak istiyorum.
      </p>
    </div>
    <div class="kayıtol-okudum">
      <input type="checkbox" />
      <p>Okudum ve onaylıyorum</p>
    </div>
    <button @click="register" class="kayıtol-devamet">
      DEVAM ET
    </button>

    <p
      style="
            text-align: center;
            color: rgba(255, 255, 255, 0.815);
            font-size: 12px;
          "
    >
      Gizlilik Politikası
    </p>
    <p
      style="
            color: rgba(244, 244, 244, 0.5);
            text-align: center;
            font-size: 12px;
          "
    >
      Epic Games hesabın var mı? Giriş Yap
    </p>
    <p
      style="
            color: rgba(244, 244, 244, 0.5);
            text-align: center;
            font-size: 12px;
            margin-top: -5px;
          "
    >
      Tüm kaydolma seçeneklerine geri dön
    </p>
    <button @click="aUpdateUser" class="updater" style="display:none"></button>
  </div>
</template>

<script>
import { mapState, mapActions } from "vuex";
import config from "../assets/config.js";
import firebase from "firebase/app";
require("firebase/auth");
export default {
  name: "kayıtol",
  data() {
    return {
      email: "",
      password: "",
      usern: null,
      success: false
    };
  },
  created() {
    this.startfirebase();
  },
  computed: {
    ...mapState(["authuser"])
  },
  methods: {
    startfirebase() {
      if (!firebase.apps.length) {
        firebase.initializeApp(config);
      } else {
        firebase.app();
      }
      firebase.auth().onAuthStateChanged(user => {
        this.usern = user;
        this.aUpdateUser(this.usern);
      });
    },
    ...mapActions(["aUpdateUser"]),

    register() {
      firebase
        .auth()
        .createUserWithEmailAndPassword(this.email, this.password)
        .then(() => {
          setTimeout(() => {
            window.location.href = "/signin";
          }, 1000);
        })
        .catch(function(error) {
          // Handle Errors here.
          var errorCode = error.code;
          var errorMessage = error.message;
          if (errorCode == "auth/weak-password") {
            alert("The password is too weak.");
          } else {
            alert(errorMessage);
          }
        });
      console.log(this.authuser);
    }
  },
  mounted() {
    let recaptchaScript = document.createElement("script");
    recaptchaScript.setAttribute(
      "src",
      "https://www.gstatic.com/firebasejs/8.2.1/firebase-app.js"
    );
    document.head.appendChild(recaptchaScript);
  }
};
</script>
