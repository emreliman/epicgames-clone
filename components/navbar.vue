<template>
  <div id="nav-bar">
    <ul>
      <nuxt-link :to="{ name: 'index' }">
        <li><img src="@/assets/epicgames-logo.png" alt="loho" /></li>
      </nuxt-link>
      <nuxt-link :to="{ name: 'mağaza' }">
        <li>MAĞAZA</li>
      </nuxt-link>
      <li>HABERLER</li>
      <li>SSS</li>
      <li>YARDIM</li>
      <li>UNREAL ENGİNE</li>
      <li>EPİC GAMES'İ İNDİR</li>
      <nuxt-link v-if="!isLogin" :to="{ name: 'kayıt0l' }">
        <li id="papara">Kayıt ol</li>
      </nuxt-link>
      <nuxt-link v-if="!isLogin" :to="{ name: 'signin' }">
        <li>Giriş Yap</li>
      </nuxt-link>
      <li v-if="isLogin">{{ email }}</li>
      <li v-if="isLogin" @click="signOut">Çıkış Yap</li>
      <li><i class="fas fa-globe"></i></li>
    </ul>
  </div>
</template>

<script>
import config from "../assets/config.js";
import firebase from "firebase/app";
require("firebase/auth");
import { mapState, mapGetters } from "vuex";
export default {
  name: "navbar",
  data() {
    return {
      isLogin: false,
      email: null
    };
  },

  methods: {
    signOut() {
      firebase.auth().signOut();
      this.isLogin = false;
      this.email = null;
    },
    // flogin(){
    //   this.isLogin = this.$store.getters["getauser"] ? true : false

    // },
    takelog() {
      this.$root.$on("logg", text => {
        this.isLogin = true;
        this.email = text;
      });
    }
  },

  created() {
    //this.flogin()
  },
  mounted() {
    this.takelog();
  }
};
</script>

<style scoped>
#papara {
  margin-left: 45%;
}
</style>
