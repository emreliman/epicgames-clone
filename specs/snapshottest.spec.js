import { mount,config,RouterLinkStub } from "@vue/test-utils";
import deneme from '@/deneme'
import smallcard from '@/smallcard'
import kayıtol from '@/kayıtol'
import girişyap from '@/girişyap'
import filteritem from '@/filteritem'
import navbar from '@/navbar'
import reqitem from '@/reqitem'
import siparişcard from '@/siparişcard'





test('Deneme ', () => {
    const wrapper=mount(deneme)
    expect(wrapper.html()).toMatchSnapshot()
})

test('smallcard ', () => {
    const smallcardwrapper=mount(smallcard,{
        propsData:{
            item:{
                "key": 1,
                "name": "AutoChess",
                "corporation":"Dragonest Game",
                "long":"Auto Chess, Dragonest Co.Ltd ve Drodo Studio tarafından ortaklaşa geliştirilen ve Dragonest Co.Ltd yayınlanan orijinal otomatik savaş oyunudur.Auto Chess çıkışından bu yana yepyeni oynanış modu ve stratejik rekabetçi özellikleriyle tüm dünyada önemli bir etki bıraktı. Ayrıca tüm dünyada milyonlarca oyuncunun ilgisini çekti. Her bir oyuncu İnternet üzerinden dünyanın dört bir köşesinden katılan başka oyunculara karşı oynayabilir.",
                "about":
                  "Auto Chess 1.0 çıktı! En iyi çevrimiçi rekabetçi strateji oyununda parçalarını seçerek ve takımını[...]",
                "img":"autochess",
                "listefiyatı":200.00,
              "indirim":50.00,
              "epickupon":50.00,
              "ücret":100.00
              }
        }
    })
    expect(smallcardwrapper.html()).toMatchSnapshot()
    
})

test('kayıtol ', () => {
    const kayıtolwrapper=mount(kayıtol)
    const mail =kayıtolwrapper.vm.email
    kayıtolwrapper.setData({email:"emreliman@gmail.com"})
    expect(kayıtolwrapper.html()).toMatchSnapshot()
})



test('girişyap ', () => {
    const gwrapper=mount(girişyap)
    expect(gwrapper.html()).toMatchSnapshot()
    
})

test('filteritem ', () => {
    const fwrapper=mount(filteritem)
    expect(fwrapper.html()).toMatchSnapshot()
})

test('navbar ', () => {
    const nwrapper=mount(navbar,{
        stubs:{
            NuxtLink:true
        }
    })
    expect(nwrapper.html()).toMatchSnapshot()
})

test('reqitem ', () => {
    const rwrapper=mount(reqitem)
    expect(rwrapper.html()).toMatchSnapshot()
})

// test.only('siparişcard ', () => {
//     const swrapper=mount(siparişcard,{
//         propsData:{
//             item:{
//                 "key": 1,
//                 "name": "AutoChess",
//                 "corporation":"Dragonest Game",
//                 "long":"Auto Chess, Dragonest Co.Ltd ve Drodo Studio tarafından ortaklaşa geliştirilen ve Dragonest Co.Ltd yayınlanan orijinal otomatik savaş oyunudur.Auto Chess çıkışından bu yana yepyeni oynanış modu ve stratejik rekabetçi özellikleriyle tüm dünyada önemli bir etki bıraktı. Ayrıca tüm dünyada milyonlarca oyuncunun ilgisini çekti. Her bir oyuncu İnternet üzerinden dünyanın dört bir köşesinden katılan başka oyunculara karşı oynayabilir.",
//                 "about":
//                   "Auto Chess 1.0 çıktı! En iyi çevrimiçi rekabetçi strateji oyununda parçalarını seçerek ve takımını[...]",
//                 "img":"autochess",
//                 "listefiyatı":200.00,
//               "indirim":50.00,
//               "epickupon":50.00,
//               "ücret":100.00
//               }
//         },
//         stubs:{
//             NuxtLink:RouterLinkStub
//         }

//     })
//     expect(swrapper.html()).toMatchSnapshot()
// })

