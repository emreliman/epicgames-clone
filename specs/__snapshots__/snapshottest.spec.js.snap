// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deneme  1`] = `<h1>Emre</h1>`;

exports[`filteritem  1`] = `
<div class="dropdown">
  <p>Filtreler</p>
  <hr> <button class="dropbtn">
    Kategoriler <i class="arrow up"></i></button>
  <div class="dropdown-content"><button href="#">Aksiyon</button> <button href="#">Macera</button> <button>Mod Araçları</button> <button>Bulmaca</button> <button>Yarış</button> <button>RYO</button> <button>Nişancı</button></div>
  <hr>
</div>
`;

exports[`girişyap  1`] = `
<div class="giriş-card">
  <div class="giriş-logo"><img src="@/assets/epicgames-logo.png" alt="logo"></div>
  <p style="color: rgba(255, 255, 255, 0.75); margin-left: 30px; font: sans-serif; font-size: 13px;">
    BİR EPIC GAMES HESABI ILE GIRIŞ YAP
  </p>
  <div class="giriş-görünenad"><input placeholder="*E-posta Adresi" required="required">
    <p style="margin-top: 25px; transform: translateX(-55px);">Gerekli</p>
  </div>
  <div class="giriş-şifre"><input placeholder="*Şifre " type="password" required="required">
    <p style="margin-top: 25px; transform: translateX(-55px);">Gerekli</p>
  </div>
  <!---->
  <div class="giriş-teklifler"><input type="checkbox">
    <p style="margin-top: 18px; margin-right: 110px;">Beni hatırla</p>
    <p style="margin-top: 18px;">Şifreni Mİ Unuttun?</p>
  </div> <button class="giriş-devamet">HEMEN GİRİŞ YAP</button>
  <p style="text-align: center; color: rgba(255, 255, 255, 0.815); font-size: 12px;">
    Gizlilik Politikası
  </p>
  <p style="color: rgba(244, 244, 244, 0.5); text-align: center; font-size: 12px;">
    Epic Games hesabın yok mu? Kayıt Ol
  </p>
  <p style="color: rgba(244, 244, 244, 0.5); text-align: center; font-size: 12px; margin-top: -5px;">
    Tüm kaydolma seçeneklerine geri dön
  </p>
</div>
`;

exports[`kayıtol  1`] = `
<div class="kayıtol-card">
  <div class="kaytol-logo"><img src="@/assets/epicgames-logo.png" alt="logo"></div>
  <div class="kayıtol-droplist">
    <p style="color: rgb(244, 244, 244); font-size: 12px; font-family: Brutal, sans-serif; margin-left: 32px;">
      KAYIT OL
    </p> <button class="kayıtol-dropbtn">
      Turkey
      <svg focusable="false" viewBox="0 0 24 24" aria-hidden="true" class="MuiSvgIcon-root" style="width: 2rem; height: 2rem; float: right;">
        <path d="M7 10l5 5 5-5z"></path>
      </svg></button>
    <div class="kayıtol-drop-content"></div>
  </div>
  <div class="kayıtol-adsoyad"><input placeholder="*Ad" required="required"> <input placeholder="*Soyadı" required="required"></div>
  <div class="kayıtol-görünenad"><input placeholder="*Görünen Ad" required="required"> <input placeholder="*E-posta Adresi" required="required"></div>
  <div class="kayıtol-şifre"><input placeholder="*Şifre " type="password" required="required"></div>
  <div class="kayıtol-teklifler"><input type="checkbox">
    <p>
      Epic Games'ten haberler, anketler ve özel teklifler almak istiyorum.
    </p>
  </div>
  <div class="kayıtol-okudum"><input type="checkbox">
    <p>Okudum ve onaylıyorum</p>
  </div> <button class="kayıtol-devamet">
    DEVAM ET
  </button>
  <p style="text-align: center; color: rgba(255, 255, 255, 0.815); font-size: 12px;">
    Gizlilik Politikası
  </p>
  <p style="color: rgba(244, 244, 244, 0.5); text-align: center; font-size: 12px;">
    Epic Games hesabın var mı? Giriş Yap
  </p>
  <p style="color: rgba(244, 244, 244, 0.5); text-align: center; font-size: 12px; margin-top: -5px;">
    Tüm kaydolma seçeneklerine geri dön
  </p> <button class="updater" style="display: none;"></button>
</div>
`;

exports[`navbar  1`] = `
<div id="nav-bar">
  <ul>
    <nuxtlink-stub to="[object Object]">
      <li><img src="@/assets/epicgames-logo.png" alt="loho"></li>
    </nuxtlink-stub>
    <nuxtlink-stub to="[object Object]">
      <li>MAĞAZA</li>
    </nuxtlink-stub>
    <li>HABERLER</li>
    <li>SSS</li>
    <nuxtlink-stub to="[object Object]">
      <li>SEPETE GİT</li>
    </nuxtlink-stub>
    <li>EPİC GAMES'İ İNDİR</li>
    <nuxtlink-stub to="[object Object]">
      <li id="papara">Kayıt ol</li>
    </nuxtlink-stub>
    <nuxtlink-stub to="[object Object]">
      <li>Giriş Yap</li>
    </nuxtlink-stub>
    <!---->
    <!---->
    <li><i class="fas fa-globe"></i></li>
  </ul>
</div>
`;

exports[`reqitem  1`] = `
<div class="req">
  <p style="color: white;">Özellikler</p>
  <div class="req-card">
    <div class="req-os" style="color: white; margin-bottom: 50px;">
      Windows
      <hr>
    </div>
    <div class="req-min-ön">
      <div class="req-min">
        <p style="color: grey; margin-bottom: 50px;">Minimum</p>
        <p style="color: grey;">İşletim Sistemi</p>
        <p style="color: white;">Windows 10 64-bit</p>
        <p style="color: grey;">İŞLEMCİ</p>
        <p style="color: white;">Core i3 3.3 Ghz</p>
        <p style="color: grey;">Bellek</p>
        <p style="color: white;">4GB RAM</p>
        <p style="color: grey;">Ekran Kartı</p>
        <p style="color: white;">Intel HD 4000</p>
      </div>
      <div class="req-ön">
        <p style="color: grey; margin-bottom: 50px;">Önerilen</p>
        <p style="color: grey;">İşletim Sistemi</p>
        <p style="color: white;">Windows 10 64-bit</p>
        <p style="color: grey;">İŞLEMCİ</p>
        <p style="color: white;">Core i5 3.5 Ghz</p>
        <p style="color: grey;">Bellek</p>
        <p style="color: white;">8 GB RAM</p>
        <p style="color: grey;">Ekran Kartı</p>
        <p style="color: white;">Nvidia GTX 660</p>
      </div>
    </div>
    <div class="req-supported-lang">
      <p style="color: grey;">Desteklenen Diller:</p>
      <p style="color: white;">İngilizce, Fransızca, Almanca, Türkçe</p>
    </div>
  </div>
</div>
`;

exports[`smallcard  1`] = `
<div class="sm-card">
  <p class="açıklama">
    Auto Chess, Dragonest Co.Ltd ve Drodo Studio tarafından ortaklaşa geliştirilen ve Dragonest Co.Ltd yayınlanan orijinal otomatik savaş oyunudur.Auto Chess çıkışından bu yana yepyeni oynanış modu ve stratejik rekabetçi özellikleriyle tüm dünyada önemli bir etki bıraktı. Ayrıca tüm dünyada milyonlarca oyuncunun ilgisini çekti. Her bir oyuncu İnternet üzerinden dünyanın dört bir köşesinden katılan başka oyunculara karşı oynayabilir.
  </p>
  <div class="sm-card-buy">
    <div class="sm-prices">
      <p id="sm-discount">
        -50%
      </p>
      <p>100₺</p>
      <p>100₺</p>
    </div>
    <div class="buy-fav-button"><button>ŞİMDİ SATIN AL</button> <button class="heart-icon"><i class="far fa-heart"></i></button></div>
    <p>indirim 05.12.2020 saat 19:00'da bitiyor.</p>
  </div>
</div>
`;
